lib/puppet/provider/package/yum.rb

```diff
@@ -142,13 +142,24 @@ def self.update_command
     'update'
   end
 
+  def makecache_fast
+    command = [command(:cmd), :makecache, :fast]
+    execute(command, :failonfail => false, :combine => false)
+  end
+
   def install
     wanted = @resource[:name]
     error_level = self.class.error_level
     update_command = self.class.update_command
     # If not allowing virtual packages, do a query to ensure a real package exists
     unless @resource.allow_virtual?
-      execute([command(:cmd), '-d', '0', '-e', error_level, '-y', install_options, :list, wanted].compact)
+      command = [command(:cmd), '-d', '0', '-e', error_level, '-y', install_options, :list, wanted].compact
```

Comment:
Sure. I had only duplicated it since I saw that in other package providers (e.g. dpkg.rb).